packages/cardano-services/src/Asset/types.ts

```diff
@@ -16,3 +16,15 @@ export interface NftMetadataService {
    */
   getNftMetadata(asset: AssetPolicyIdAndName): Promise<Asset.NftMetadata | undefined>;
 }
+
+/**
+ * Service to get CIP-38? token metadata for a given subject
+ */
+export interface TokenMetadataService {
```

Comment:
While working on ADP-1606 I have a doubt, could it be better to call this `TokenMetadataProvider` ?

Reply:
Probably best to leave it as-is for now. All "Provider" interfaces are currently specified in `core` package. When we write docs we need to come up with a good definition of what exactly a cardano-js-sdk Provider means.

Reply:
The naming is intentionally aligned with https://github.com/input-output-hk/cardano-js-sdk/blob/6431f57ee5d0624784e9ae6c6b6a73b35a514ee0/packages/cardano-services/src/Metadata/types.ts#L3-L5